it/src/main/java/com/google/cloud/teleport/it/bigtable/BigtableResourceManager.java

```diff
@@ -0,0 +1,90 @@
+   *     given Bigtable instance.
+   * @throws RuntimeException if there is an error creating the instance in Bigtable.
+   */
+  void createInstance(Iterable<BigtableResourceManagerCluster> clusters)
+      throws IllegalStateException;
+
+  /**
+   * Creates a table within the current instance given a table ID.
+   *
+   * <p>Note: Implementations may do instance creation here, if one does not already exist.
+   *
+   * @param tableId The id of the table.
+   * @param columnFamilies A collection of column family names for the table.
+   * @throws RuntimeException if there is an error creating the table in Bigtable.
+   */
+  void createTable(String tableId, Iterable<String> columnFamilies) throws IllegalStateException;
+
+  /**
+   * Writes a given row into a table. This method requires {@link
+   * BigtableResourceManager#createTable(String, Iterable)} to be called for the target table
+   * beforehand.
+   *
+   * @param tableRow A mutation object representing the table row.
+   * @throws RuntimeException if method is called after resources have been cleaned up, if the
+   *     manager object has no instance, if the table does not exist or if there is an IOException
+   *     when attempting to retrieve the bigtable data client.
+   */
+  void write(RowMutation tableRow) throws IllegalStateException;
+
+  /**
+   * Writes a collection of table rows into one or more tables. This method requires {@link
+   * BigtableResourceManager#createTable(String, Iterable)} to be called for the target table
+   * beforehand.
+   *
+   * @param tableRows A collection of mutation objects representing table rows.
+   * @throws RuntimeException if method is called after resources have been cleaned up, if the
+   *     manager object has no instance, if the table does not exist or if there is an IOException
+   *     when attempting to retrieve the bigtable data client.
+   */
+  void write(Iterable<RowMutation> tableRows) throws IllegalStateException;
+
+  /**
+   * Reads all the rows in a table. This method requires {@link
+   * BigtableResourceManager#createTable(String, Iterable)} to be called for the target table
+   * beforehand.
+   *
+   * @param tableId The id of table to read rows from.
+   * @return A List object containing all the rows in the table.
+   * @throws IllegalStateException if method is called after resources have been cleaned up, if the
+   *     manager object has no instance, if the table does not exist or if there is an IOException
+   *     when attempting to retrieve the bigtable data client.
+   */
+  ImmutableList<Row> readTable(String tableId) throws IllegalStateException;
+
+  /**
+   * Deletes all created resources (instance and tables) and cleans up all Bigtable clients, making
+   * the manager object unusable.
+   *
+   * @throws IllegalStateException if there is an error deleting the instance or tables in Bigtable.
+   */
+  void cleanup() throws IllegalStateException;
```

Comment:
Just a nitpick, other ResourceManagers have/are being created with `cleanupAll()`

Reply:
I was contemplating renaming the spanner RM to use cleanup() along with the BigQuery RM I'm working on, but I went ahead and refactored to cleanupAll() as suggested.